lib/peep.rb

```diff
@@ -0,0 +1,45 @@
+require "./lib/select_database"
+require "./lib/database_connection"
+
+class Peep 
+
+  attr_reader :id, :body, :posted_at, :author
+
+  def initialize(id, body, date, user_id)
+    @id = id
+    @body = body
+    @posted_at = date
+    @user_id = user_id
+    @author = find_author
+  end
+
+  def self.all
+    select_database
+
+    peeps = DatabaseConnection.query("SELECT * FROM peeps ORDER BY id DESC")
+
+    peeps.map do |peep|
+      Peep.new(peep["id"], peep["body"], peep["posted_at"], peep["user_id"])
+    end
+  end
+
+  def self.add(peep, user_id)
```

Comment:
Minor: I would use 'body' or 'text' or 'peep_text' (etc etc) here, instead of just peep. My expectation when I see the variable peep here is that it is an instead of Peep, not just the body.

Similarly, in the method above, I might be tempted to use peeps_data and peeps_data.map do |peep_data|, just to disambiguate the data from the model.